upgrade/1.2/scripts/ceph/lib/ceph-health.sh

```diff
@@ -120,3 +120,55 @@ function wait_for_osds() {
    done
  done
 }
+
+function get_ip_from_metadata() {
+  host=$1
+  ip=$(cloud-init query ds | jq -r ".meta_data[].host_records[] | select(.aliases[]? == \"$host\") | .ip" 2>/dev/null)
+  echo $ip
+}
+
+function wait_for_mon_stat() {
+  node=$1
+  cnt=0
+  while true; do
+    if [[ "$cnt" -eq 60 ]]; then
+      echo "ERROR: Giving up waiting for mon process to start on $node..."
+      break
+    fi
+    if [[ "$cnt" -eq 30 ]]; then
+      echo "Manually adding mon process for $node..."
+      ip=$(get_ip_from_metadata "${node}.nmn")
+      ceph mon add $node $ip
+    else
+      state_name=$(ceph mon stat -f json-pretty | jq --arg node $node -r '.quorum[] | select(.name==$node) | .name' )
+      if [ "$state_name" == "$node" ]; then
+        echo "Found mon process on $node..."
+        break
+      fi
+    fi
+    sleep 5
+    echo "Sleeping for five seconds waiting for mon process to start on $node..."
+    cnt=$((cnt+1))
+  done
+}
+
+function check_mon_daemon() {
+  node=$1
+  state_name=$(ceph mon stat -f json-pretty | jq --arg node $node -r '.quorum[] | select(.name==$node) | .name' )
+  if [ "$state_name" == "$node" ]; then
+    echo "Found ${node} in ceph mon stat command, continuing..."
+  else
+    echo "Didn't find ${node} in ceph mon stat command, ensuring we have quorum before restarting daemon..."
```

Comment:
we should test stopping the pod and waiting for 5-10 mins and see if it leaves the stat command.  I don't believe it will as the daemon hasn't been removed, but just to protect against us not starting a mon that we should be starting.

Reply:
Stopping the podman container indeed (within abt 5 secs) removed the node name from the stat command.  Left it for 10 mins, orch output still showed the daemon running.  Then removing the daemon indeed restarted the daemon, podman process, and the node re-appeared in the stat command -- so looks like the code handled this case..